### PR TITLE
Fix branch list include commit id

### LINKED ISSUE #1216: List branches should return branch ID and Commit
Regression - list branches should list the branch names with the last commit ID branch.
Currently it lists just the branch names.

### PR DESCRIPTION
**Pull request by nopcoder**
Fix #1216

List branches API, UI, and CLI will list the names of the branch and the last commit ID on each branch.

### CODE REVIEW

File: nessie/sanity_api_test.go [hunk 1]
```diff
@@ -73,8 +73,18 @@ func TestSanityAPI(t *testing.T) {
...
 		nil)
 	require.NoError(t, err, "list branches")
-	require.ElementsMatch(t, branchesResp.Payload.Results, []string{masterBranch, "import-from-inventory", "branch1"},
+	var branches []string
+	for _, ref := range branchesResp.Payload.Results {
+		branches = append(branches, swag.StringValue(ref.ID))
+	}
+	require.ElementsMatch(t, branches, []string{masterBranch, "import-from-inventory", "branch1"},
 		"match existing branches")
+	for _, ref := range branchesResp.Payload.Results {
```
Comment (guy-har) [inline nessie/sanity_api_test.go:82]: Why not do this together?

File: cmd/lakectl/cmd/branch.go [hunk 2]
```diff
@@ -43,15 +43,15 @@ var branchListCmd = &cobra.Command{
...
 
 		ctx := struct {
 			BranchTable *Table
 			Pagination  *Pagination
 		}{
 			BranchTable: &Table{
-				Headers: []interface{}{"Branch"},
+				Headers: []interface{}{"Branch", "Commit ID"},
 				Rows:    rows,
 			},
```
Comment (guy-har) [inline cmd/lakectl/cmd/branch.go:56]: why did we add commit ID to branch log?

File: catalog/mvcc/cataloger_list_branches.go [hunk 3]
```diff
@@ -25,15 +25,25 @@ func (c *cataloger) ListBranches(ctx context.Context, repository string, prefix
...
+		if err := tx.Select(&branchesNameAndCommit, query, repoID, prefixCond, after, limit+1); err != nil {
 			return nil, err
 		}
+		branches := make([]*catalog.Branch, 0, len(branchesNameAndCommit))
+		for _, nc := range branchesNameAndCommit {
+			branches = append(branches, &catalog.Branch{
+				Name:      nc.Name,
+				Reference: MakeReference(nc.Name, CommitID(nc.CommitID)),
+			})
+		}
```
Comment (guy-har) [inline catalog/mvcc/cataloger_list_branches.go:46]:
```suggestion
		branches := make([]*catalog.Branch, 0, len(branchesNameAndCommit))
		for i, nc := range branchesNameAndCommit {
			branches[i] = &catalog.Branch{
				Name:      nc.Name,
				Reference: MakeReference(nc.Name, CommitID(nc.CommitID)),
			}
		}
```

**Review by guy-har — APPROVED** [review]
LGTM

File: nessie/sanity_api_test.go (hunk 1 above)
Comment (nopcoder) [reply nessie/sanity_api_test.go:82]: merged

File: cmd/lakectl/cmd/branch.go (hunk 2 above)
Comment (nopcoder) [reply cmd/lakectl/cmd/branch.go:56]: did some digging - https://github.com/treeverse/lakeFS/commit/548e1d1d8f19f6ad05390fd93d610a773708b603#diff-d57f018b31eac5e4a0a201b50e3f2f5d0c193e682719b5ebb65bb612500991e7R56

File: catalog/mvcc/cataloger_list_branches.go (hunk 3 above)
Comment (nopcoder) [reply catalog/mvcc/cataloger_list_branches.go:46]: tl;dr updated the code
1. my version doesn't do re-allocation - we made sure the capacity is set to the number of elements we append
2. updated the code to a version we preallocate and set by index
3. in your suggestion you need to `make([]*catalog.Branch, len(branchesNameAndCommit))` to make it work

**Pull request merged**